assets/js/search.js

```diff
@@ -0,0 +1,218 @@
+ */
+
+(function () {
+
+  // Returns a function, that, as long as it continues to be invoked, will not
+  // be triggered. The function will be called after it stops being called for
+  // N milliseconds. If `immediate` is passed, trigger the function on the
+  // leading edge, instead of the trailing.
+  // From: https://davidwalsh.name/javascript-debounce-function
+  function debounce(func, wait, immediate) {
+    let timeout;
+    return function () {
+      let context = this,
+        args = arguments;
+      let later = function () {
+        timeout = null;
+        if (!immediate) func.apply(context, args);
+      };
+      let callNow = immediate && !timeout;
+      clearTimeout(timeout);
+      timeout = setTimeout(later, wait);
+      if (callNow) func.apply(context, args);
+    };
+  }
+
+  function showItemsCount(totalCount, numSubmodules) {
+    if (searchEntry.type == 'libraryEntries') {
+      if (totalCount > 0 && numSubmodules > 0) {
+        $('#search-results-count').html("<strong>" + totalCount + "</strong> repos (~<strong>" + numSubmodules + "</strong> modules)");
+      } else {
+        $('#search-results-count').text("0 repos");
+      }
+    } else {
+      if (totalCount > 0 && numSubmodules == 0) {
+        $('#search-results-count').html("<strong>" + totalCount + "</strong> found");
+      }
+    }
+  }
+
+  function displayFilterTags() {
+    if (searchEntry.type == 'guideEntries') {
+
+      //Filters the tags from the array of objects and flattens it out since it returns an array of arrays
+      let tags = searchEntry.entries.map(entry => entry.tags.split(',')).reduce((a, b) => a.concat(b), []);
+
+      tags.filter((tag, index) => {
+        if ((tags.indexOf(tag) === index) && (tag.length != "")) {
+          $('.tags').append(`<label class="checkbox"><input value=${tag} type="checkbox">${tag}</label>`);
+        }
+      });
+    }
+  }
+
+  function showInitialItemsCount() {
+    let numSubmodules = 0;
+    for (let i = 0; i < searchEntry.length; i++) {
+      if (searchEntry.type == 'libraryEntries') return numSubmodules += libraryEntries[i].num_submodules;
+      return searchEntry;
+    }
+    showItemsCount(searchEntry.length, numSubmodules);
+  }
+
+  // Show initial module count on load
+  let searchEntry = detectSearchEntry();
+
+  $(showInitialItemsCount);
+  $(displayFilterTags);
+
+  $('#no-matches').hide();
+
+  function detectSearchEntry() {
+    let entries = [];
+    if (window.libraryEntries) {
+
+      entries = window.libraryEntries;
+      return {
+        entries,
+        type: 'libraryEntries'
+      };
+
+    } else if (window.guideEntries) {
+
+      entries = window.guideEntries;
+      return {
+        entries,
+        type: 'guideEntries'
+      };
+    }
+  }
+
+  //Function to track Google analytics
+  // function trackGA {
+  //    // Track what users are searching for via Google Analytics events
+  //     //   ga('send', 'event', "iac-lib", "search", "query", text);
+  // }
```

Comment:
?

Reply:
This one looks like it's a TO-DO mentioned in the PR description